CMakeLists.txt

```diff
@@ -384,14 +327,147 @@ function( target_link_exe_with_core target )
+	    if (debug)
+		    target_include_directories( ${testName} BEFORE PRIVATE ${CMAKE_SOURCE_DIR}/include/debug )
+		    target_link_libraries(${testName} lpf_debug lpf_hl_debug GTest::gtest GTest::gtest_main)
+	    else(debug)
+		    target_link_libraries(${testName} GTest::gtest GTest::gtest_main)
+	    endif(debug)
+
+
+	    # Extract test-specific information from comments of tests
+	    file(READ ${testSource} fileContents)
+	    string(REGEX MATCH "Exit code: ([0-9]+)" _ ${fileContents})
+	    set(retCode ${CMAKE_MATCH_1})
+	    string(REGEX MATCH "pre P >= ([0-9]+)"  _ ${fileContents})
+	    set(minProcs ${CMAKE_MATCH_1})
+	    string(REGEX MATCH "pre P <= ([0-9]+)" _ ${fileContents})
+	    set(maxProcs ${CMAKE_MATCH_1})
+	    string(REGEX MATCH "-probe ([0-9]+.[0-9]+)" _ ${fileContents})
+	    set(lpfProbeSecs ${CMAKE_MATCH_1})
+
+	    # The "\pre P <= max" comment in a test indicates the desired number of
+	    # maximum LPF processes. If, at CMake configuration, the number of detected
+	    # processes is smaller, it overrides the desired number of maximum LPF
+	    # processes. If the test does not define a desired number of maximum LPF
+	    # processes, it will be set to the detected number of processes.
+	    #
+	    # The "\pre P >= min" comment in a test indicates the desired number of
+	    # minimum LPF processes. If the number of detected processes is smaller, it
+	    # overrides the desired number of minimum LPF processes. If the test does
+	    # not define a desired minimum number of LPF processes, it will be set to
+	    # the detected number of processes.
+	    #
+	    # Hence if neither a desired min nor max number of processes are defined in
+	    # a test, then the test will execute using the number of detected
+	    # processes only.
+	    #
+	    # Most tests only define a mininum number of desired processes, such as
+	    # "\pre P >= 1". In those cases, the test will execute for all possible
+	    # number of processes between 1 and the detected number of processes. This
+	    # has been the behaviour of the test suite since v1.0.
+
+	    target_link_exe_with_core(${testName} ${ENGINE})
+
+
+	    ProcessorCount(nprocs)
+	    if ("${minProcs}" STREQUAL "")
+		    ProcessorCount(minProcs)
+	    endif()
+	    if ("${maxProcs}" STREQUAL "")
+		    ProcessorCount(maxProcs)
+	    endif()
+	    if ("${minProcs}" GREATER "${nprocs}")
+		    set(minProcs ${nprocs})
+	    endif()
+	    if ("${maxProcs}" GREATER "${nprocs}")
+		    set(maxProcs ${nprocs})
+	    endif()
+	    if ("${lpfProbeSecs}" STREQUAL "")
+		    set(lpfProbeSecs "0.0")
+	    endif()
+	    if ("${retCode}" STREQUAL "")
+		    set(retCode "0")
+	    endif()
+
+	    # Most recent approach to Gtests, recommended!
+	    set_property(TARGET ${testName} PROPERTY TEST_LAUNCHER ${MY_TEST_LAUNCHER};-e;${ENGINE};-L;${CMAKE_BINARY_DIR}/lpfrun_build;-p;${minProcs};-P;${maxProcs};-t;${lpfProbeSecs};-R;${retCode})
```

Comment:
i'd prefer the long names for the test_launcher options.  The short hands "-e" or "-P" are kind of cryptic. Because the test_launcher.py has nice long names for that like "--engine" and ""--max_processor_count", I think this line here becomes much more readable if the long ones are used instead

Reply:
Done.